Doppler.MercadoPagoApi/Controllers/PaymentController.cs

```diff
@@ -1,15 +1,65 @@
+using Doppler.MercadoPagoApi.DopplerSecurity;
+using Doppler.MercadoPagoApi.Models;
+using Doppler.MercadoPagoApi.Services;
+using MercadoPago.Client.Payment;
+using MercadoPago.Error;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using System.Net;
+using System.Threading.Tasks;
 
 namespace Doppler.MercadoPagoApi.Controllers
 {
-    [Authorize]
     [ApiController]
-    public class PaymentController : Controller
+    public class PaymentController : ControllerBase
     {
-        public PaymentController()
+        private readonly IMercadoPagoService _mercadoPagoService;
+        private readonly IMessageHandler _messageHandler;
+
+        public PaymentController(IMercadoPagoService mercadoPagoService, IMessageHandler messageHandler)
+        {
+            _mercadoPagoService = mercadoPagoService;
+            _messageHandler = messageHandler;
+        }
+
+        [Authorize(Policies.OWN_RESOURCE_OR_SUPERUSER)]
+        [HttpPost("/accounts/{accountname}/payment")]
+        public async Task<IActionResult> Post([FromRoute] string accountname, [FromBody] PaymentRequestDto paymentRequestDto)
         {
+            try
+            {
+                var cardToken = await _mercadoPagoService.CreateTokenAsync(paymentRequestDto.Card);
+
+                var paymentRequestCreated = new PaymentCreateRequest
+                {
+                    TransactionAmount = paymentRequestDto.TransactionAmount,
+                    Token = cardToken.Id,
+                    Installments = paymentRequestDto.Installments,
+                    PaymentMethodId = paymentRequestDto.PaymentMethodId,
+                    Payer = new PaymentPayerRequest
+                    {
+                        Id = paymentRequestDto.Payer.Id,
+                        Email = accountname,
+                        Type = paymentRequestDto.Payer.Type
+                    },
+                };
+
+                var paymentResponse = await _mercadoPagoService.CreatePaymentAsync(paymentRequestCreated);
+
+                var paymentResponseDto = new PaymentResponseDto
+                {
+                    Id = paymentResponse.Id,
+                    Status = paymentResponse.Status,
+                    StatusDetail = paymentResponse.StatusDetail,
+                    Message = _messageHandler.GetMessage(paymentResponse)
+                };
 
+                return Ok(paymentResponseDto);
+            }
+            catch (MercadoPagoApiException e)
+            {
+                return BadRequest(e.ApiError);
```

Comment:
The 400 (Bad Request) status code indicates that the request sent from the client is not properly constructed. This sounds a lot more like a 500 (Internal Server Error) because Mercado Pago API returns an error.
Anyway, I'm not sure how you should handle it, if at all. Please confirm with @andresmoschini and @jhoffman-ms

Reply:
We have already discussed it here: https://github.com/FromDoppler/doppler-mercadopago-api/pull/8/files#r868108206

I think as you @afantini, in my opinion, it should be a 500 error while we are not ready to control the error.

Reply:
We did that because when a MercadoPagoApiException is thrown, it is a BadRequest from the MercadoPago library (with errors like "invalid card number" or "invalid expiration date") . But after your and Alejandro's explanations I understand and I agree! Thanks!